src/common/FileSystem.cpp

```diff
@@ -1298,17 +1376,46 @@ std::string ReadFile(Str::StringRef path, std::error_code& err)
 		if (err)
 			return "";
 
+		bool isSymLink = zipFile.CheckSymlink(err);
+		if (err) {
+			return "";
+		}
+
 		// Close file and check for CRC errors
 		zipFile.CloseFile(err);
 		if (err)
 			return "";
 
+		if (isSymLink) {
+			if (symlinkLevels <= 0) {
+				Log::Debug("symlink: %s: too many levels of symbolic links", path);
+				return "";
+			}
+			out = expandSymlink(path, out);
+			if (out[0] == '\0') {
+				return "";
+			}
+			out = ReadFile(out, symlinkLevels -1, err);
+			if (err) {
+				return "";
+			}
+		}
+
 		return out;
 	}
 
 	ASSERT_UNREACHABLE();
 }
 
+std::string ReadFile(Str::StringRef path, std::error_code& err)
+{
+	std::string out = ReadFile(path, 16, err);
```

Comment:
named constant for this number. MAX_SYMLINK_LEVELS or something.

Reply:
yep